frontend/src/metabase/query_builder/components/template_tags/TagEditorParam.jsx

```diff
@@ -120,20 +123,10 @@ export class TagEditorParam extends Component {
       if (!field) {
         return;
       }
-      const options = getParameterOptionsForField(field);
-      let widgetType;
-      if (
-        tag["widget-type"] &&
-        _.findWhere(options, { type: tag["widget-type"] })
-      ) {
-        widgetType = tag["widget-type"];
-      } else if (options.length > 0) {
-        widgetType = options[0].type;
-      }
       setTemplateTag({
         ...tag,
         dimension,
-        "widget-type": widgetType,
+        "widget-type": getDefaultParameterWidgetType(tag, field),
```

Comment:
If everything works out I'll migrate `TagEditorParam` to typescript in the next PR